java/flight/src/main/java/org/apache/arrow/flight/FlightEndpoint.java

```diff
@@ -34,9 +36,11 @@ public FlightEndpoint(Ticket ticket, Location... locations) {
     this.ticket = ticket;
   }
 
-  public FlightEndpoint(Flight.FlightEndpoint flt) {
-    locations = flt.getLocationList().stream()
-        .map(t -> new Location(t)).collect(Collectors.toList());
+  public FlightEndpoint(Flight.FlightEndpoint flt) throws URISyntaxException {
+    locations = new ArrayList<>();
+    for (final Flight.Location location : flt.getLocationList()) {
```

Comment:
Why the move away from streams in some of these changes?

Reply:
Ah, here it's because I wanted the checked exception to be propagated - the stream hides this.